presto-memory/src/main/java/com/facebook/presto/plugin/memory/MemoryPageSourceProvider.java

```diff
@@ -55,6 +55,7 @@ public ConnectorPageSource createPageSource(
         List<Integer> columnIndexes = columns.stream()
                 .map(MemoryColumnHandle.class::cast)
                 .map(MemoryColumnHandle::getColumnIndex).collect(toList());
+        pagesStore.initialize(tableId);
```

Comment:
this seems to be a bug as well. Please consider similar scenario scenario as in previous comment. There had been some write query and so worker stored data for this table, then there was a crash and worker got restarted. Now, with your change, when select comes worker would just return no data in the same way as it there were never any data.

And once again, I wish we had tests for that.